packages/curvenote-cli/src/session/session.ts

```diff
@@ -216,16 +128,144 @@ export class Session implements ISession {
+    this.log.debug('SessionToken: requesting a new session token.');
+    const {
+      decoded: { aud },
+    } = this.$activeTokens.user;
+    try {
+      const response = await this.fetch(aud as string, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${this.$activeTokens.user.token}`,
+        },
+      });
+      if (!response.ok) {
+        this.log.debug(`Response: ${response.status} ${response.statusText}`);
+        throw new Error(`SessionToken: The user token is not valid.`);
+      }
+      const json = (await response.json()) as { session?: string };
+      if (!json.session)
+        throw new Error(
+          "SessionToken: There was an error in the response, expected a 'session' in the JSON object.",
+        );
+      const decoded = jwt.decode(json.session) as Token['decoded'];
+      this.$activeTokens.session = { token: json.session, decoded };
+      this.log.debug('SessionToken: new session token created.');
+      this.log.debug('SessionToken payload:');
+      this.log.debug(JSON.stringify(decoded, null, 2));
+    } catch (error) {
+      this.log.error(
+        `⛔️ There was a problem with your API token or the API at ${aud} is unreachable.`,
+      );
+      this.log.error('Run `curvenote token check` to see the status of your token.');
+      this.log.error(
+        'If the error persists try generating a new token or contact support@curvenote.com.',
+      );
+      this.log.debug(chalk.red(error));
+      throw new Error('Could not refresh session token');
+    }
+  }
+
+  async getHeaders() {
+    const headers: Record<string, string> = {
+      'X-Client-Name': XClientName.javascript,
+      'X-Client-Version': CLIENT_VERSION,
+    };
+    this.refreshSessionToken();
+    if (this.$activeTokens.session) {
+      headers.Authorization = `Bearer ${this.$activeTokens.session.token}`;
+    }
+    return headers;
+  }
+
+  async configure() {
+    if (!this.$activeTokens.session?.decoded.aud) return;
+    const { aud, cfg } = this.$activeTokens.session.decoded;
+    const audience = Array.isArray(aud)
+      ? aud[0]
+      : (this.$activeTokens.session.decoded.aud as string);
+
+    if (cfg) {
+      this.log.debug(`Configure using 'cfg' claim: "${cfg}".`);
+      try {
+        const headers = await this.getHeaders();
+        this.log.debug(`GET ${cfg}`);
+        const response = await this.fetch(cfg, {
+          method: 'get',
+          headers: {
+            'Content-Type': 'application/json',
+            ...headers,
+          },
+        });
+        if (response.ok) {
+          this.log.debug(`Response ok.`);
+          this.$config = (await response.json()) as CLIConfigData;
+          this.log.debug(`Configuration set: "${JSON.stringify(this.$config, null, 2)}".`);
+
+          // We are still setting this as some of the myst-cli functions rely on it
+          this.API_URL = this.$config?.editorApiUrl ?? 'INVALID';
```

Comment:
we need to keep setting this in order to keep some of the `myst-cli` things we are using happy (templatesUrl), works for now.